src/metatrain/utils/llpr.py

```diff
@@ -589,64 +588,54 @@ def save_checkpoint(self, path: Union[str, Path]):
                 "calibrated yet."
             )
 
-        # FIXME: this is very hacky, there should be a way to get the architecture
-        # name from a model instance
-        wrapped_architecture_name = self.model.__module__.replace(
-            "metatrain.", ""
-        ).replace(".model", "")
+        wrapped_model_checkpoint = self.model._get_checkpoint()
```

Comment:
It seems that now, in order to use the LLPR, a model also needs to implement this. It should be added to the docs for the LLPR (we have a list of things to do in order to make a model LLPR-compatible) or perhaps it should be added as a method that all models need to have

Reply:
Yes, that works for me!

Reply:
@pfebrer, this will be another breaking change for the MACE branch. It should be easy to update, it is basically doing half of the job that the trainer was doing before when creating the checkpoint